contracts/swappers/async/ZRXSwapper.sol

```diff
@@ -39,20 +34,15 @@ contract ZRXSwapper is IZRXSwapper, Swapper {
     uint256, // Max slippage is used off-chain
     bytes calldata _data
   ) internal override returns (uint256 _receivedAmount) {
-    (
-      address inputToken,
-      address outputToken,
-      uint256 inputAmount, // minAmountOut
-      ,
-
-    ) = abi.decode(_data[4:], (address, address, uint256, uint256, Transformation[]));
-
-    require(inputToken == _tokenIn && outputToken == _tokenOut && inputAmount == _amountIn, 'Swapper: incorrect swap information');
     IERC20(_tokenIn).safeApprove(ZRX, 0);
     IERC20(_tokenIn).safeApprove(ZRX, _amountIn);
     (bool success, ) = ZRX.call{value: 0}(_data);
     require(success, 'Swapper: ZRX trade reverted');
+    // Check that token in & amount in was correct
+    require(IERC20(_tokenIn).balanceOf(address(this)) == 0, 'Swapper: incorrect swap information');
+    // Check that token out was correct
     _receivedAmount = IERC20(_tokenOut).balanceOf(address(this));
+    require(_receivedAmount > 0, 'Swapper: incorrect swap information');
```

Comment:
received amount should be checked with balaceOf on before and after call(swap) 
this way airdropped tokens [pre-existent balances] will mess up calculations, and potentially allow an exploit on the swap